hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -285,16 +285,29 @@ public void addTransactions(Map<Long, List<Long>> containerBlocksMap)
     lock.lock();
     try {
       ArrayList<DeletedBlocksTransaction> txsToBeAdded = new ArrayList<>();
+      long currentBatchSizeBytes = 0;
+      final long maxBatchSizeBytes = 25 * 1024 * 1024; // 25MB
       for (Map.Entry< Long, List< Long > > entry :
           containerBlocksMap.entrySet()) {
         long nextTXID = sequenceIdGen.getNextId(DEL_TXN_ID);
         DeletedBlocksTransaction tx = constructNewTransaction(nextTXID,
             entry.getKey(), entry.getValue());
         txsToBeAdded.add(tx);
+        // Rough size estimation: 8 bytes txnId + 8 bytes containerId + (block count * 8 bytes) + 2 bytes(count)
+        long txSize = 8L + 8L + tx.getLocalIDList().size() * 8L + 2L;
```

Comment:
Is there any direct way to get the tx size, instead of the estimation?

Reply:
Please add a unit test to verify this improvement.

Reply:
Now, using tx.getSerializedSize(), a unit test was added.